homeassistant/exceptions.py

```diff
@@ -21,6 +21,16 @@ def __init__(
         translation_placeholders: dict[str, str] | None = None,
     ) -> None:
         """Initialize exception."""
+        # pylint: disable-next=import-outside-toplevel
+        from .helpers.translation import async_get_exception_message
```

Comment:
We tend to throw exceptions like this quite a lot to control flow. It would be nice to avoid the late here.

Reply:
If this is moved to `__str__`, this should be less of a concern, right?